backend/driver.pm

```diff
@@ -53,41 +61,34 @@ sub new {
 sub start {
     my ($self) = @_;
 
-    my $p1, my $p2;
-    pipe($p1, $p2) or die "pipe: $!";
-    $self->{from_parent} = $p1;
-    $self->{to_child}    = $p2;
-
-    $p1 = undef;
-    $p2 = undef;
-    pipe($p1, $p2) or die "pipe: $!";
-    $self->{to_parent}  = $p2;
-    $self->{from_child} = $p1;
+    open(my $STDOUTPARENT, '>&', *STDOUT);
+    open(my $STDERRPARENT, '>&', *STDERR);
 
-    printf STDERR "$$: to_child %d, from_child %d\n", fileno($self->{to_child}), fileno($self->{from_child});
+    my $backend_process = process(sub {
+            my $process = shift;
+            $SIG{TERM} = 'DEFAULT';
+            $SIG{INT}  = 'DEFAULT';
+            $SIG{HUP}  = 'DEFAULT';
+            #  $SIG{CHLD} = 'DEFAULT';
+            $0 = "$0: backend";
 
-    my $pid = fork();
-    die "fork failed" unless defined $pid;
+            open STDOUT, ">&", $STDOUTPARENT;
+            open STDERR, ">&", $STDERRPARENT;
+            # now initialize opencv
+            require cv;
 
-    if ($pid == 0) {
-        $SIG{TERM} = 'DEFAULT';
-        $SIG{INT}  = 'DEFAULT';
-        $SIG{HUP}  = 'DEFAULT';
-        $SIG{CHLD} = 'DEFAULT';
-        $0         = "$0: backend";
+            cv::init();
+            require tinycv;
 
-        # now initialize opencv
-        require cv;
+            $self->{backend}->run(fileno($process->channel_in), fileno($process->channel_out));
+            _exit(0);
+    })->blocking_stop(1)->separate_err(0)->subreaper(1)->start;
 
-        cv::init();
-        require tinycv;
+    $backend_process->on(collected => sub { diag "backend process exited: " . shift->exit_status; });
```

Comment:
Somehow this message scapes the logger, and ends up all naked, the same as the other message above, perhaps the logger facility goes out of scope?

Reply:
Possible, have to look at that still

Reply:
Checked now.. actually it's not using bmwqemu::diag, but the one in the driver (which is naked) https://github.com/os-autoinst/os-autoinst/blob/master/backend/driver.pm#L37 . wanna use bmwqemu::diag instead? i thought we wanted to get rid of bmwqemu at some point

Reply:
Yeah, but We'd rather have it all centralized so that we don't end up with this kind of confusion :) I mean, bmwqemu::diag is already adding the stuff to the logging (maybe wrap it?). And we can remove it later (and easier) when we revamp the logging functions...

Reply:
It would be useful to know the original code location where  the call back was set as well

Reply:
@richiejp argh, missed the comment, good point